packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -331,6 +331,21 @@ export class MirrorNodeClient {
             requestId);
     }
 
+    /**
+     * In some very rare cases the /contracts/results api is called before all the data is saved in
+     * the mirror node DB and `transaction_index` is returned as `undefined`. A single re-fetch is sufficient to
+     * resolve this problem.
+     * @param transactionIdOrHash
+     * @param requestId
+     */
+    public async getContractResultWithRetry(transactionIdOrHash: string, requestId?: string) {
```

Comment:
Works for now but would be nice if we could update the retry logic to take custom check like this in addition to the HTTP codes